module/mempool/stdmap/pending_receipts.go

```diff
@@ -0,0 +1,109 @@
+package stdmap
+
+import (
+	"fmt"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// PendingReceipts stores pending receipts indexed by the id.
+// It also maintains a secondary index on the previous result id, which is unique,
+// in order to allow to find a receipt by the previous result id.
+type PendingReceipts struct {
+	*Backend
+	// secondary index by parent result id, has to be unqiue
+	byPreviousResultID map[flow.Identifier]flow.Identifier
+}
+
+func indexByPreviousResultID(receipt *flow.ExecutionReceipt) flow.Identifier {
+	return receipt.ExecutionResult.PreviousResultID
+}
+
+// NewPendingReceipts creates a new memory pool for execution receipts.
+func NewPendingReceipts(limit uint) *PendingReceipts {
+	// create the receipts memory pool with the lookup maps
+	r := &PendingReceipts{
+		Backend:            NewBackend(WithLimit(limit)),
+		byPreviousResultID: make(map[flow.Identifier]flow.Identifier),
+	}
+	r.RegisterEjectionCallbacks(func(entity flow.Entity) {
+		receipt := entity.(*flow.ExecutionReceipt)
+		previousResultID := indexByPreviousResultID(receipt)
+		delete(r.byPreviousResultID, previousResultID)
+	})
+	return r
+}
+
+// Add adds an execution receipt to the mempool.
+func (r *PendingReceipts) Add(receipt *flow.ExecutionReceipt) bool {
+	previousResultID := indexByPreviousResultID(receipt)
+	var exists bool
```

Comment:
I don't think this satisfies your API: 
https://github.com/onflow/flow-go/blob/d831cc2a1fb445126469f6f291c7cdcfea971a66/module/mempool/pending_receipts.go#L9-L12

If it is a duplicate, isn't `exists` going to be true?

Reply:
yeah, good point.